### PR TITLE
add typescript

### PR DESCRIPTION
**Pull request by yann300**
a basic test would consist of 3 files in 3 different folders importing themselves, with some files being js, some other ts, like:

```
import { test, fn } from './test'

console.log('test', test)
console.error(fn())
```

```
import * as test2 from './subfolder/test2.js'
export const test = 34;

export const fn = () => {
    return test2.fn2();
}
```

```
export const test2 = 34;

export const fn2 = () => {
    return 445;
}
```

### CODE REVIEW

File: src/script-runner.js [hunk 1]
```diff
@@ -16,24 +17,72 @@ window.starknet = starknet
...
+        return returns
       } catch (e) {
         this.emit('error', {
           data: [e.message]
         })
       }
     }
   }
+
+  async _resolveFile (fileName) {
```
Comment (ioedeveloper) [inline src/script-runner.js:72]: With this implementation, i will not be able to have two files of the same name with different extensions in the same directory. E.g `script.js` and `script.ts`. Is this the intended implementation?

**Review by Aniket-Engg — COMMENTED** [review]
If I import a file starting with `../`, it shows error: `Error from IDE : Error: Cannot read/write to path outside workspace`

**Comment by yann300** [comment]
> If I import a file starting with ../, it shows error: Error from IDE : Error: Cannot read/write to path outside workspace

fixed

File: src/script-runner.js (hunk 1 above)
Comment (yann300) [reply src/script-runner.js:72]: I think this is ok if we set the `ts` file to be run instead of `js`

**Review by Aniket-Engg — APPROVED** [review]
Looks fine but import without assignment show issues. This can be fixed in different PR. Track it: https://github.com/yann300/remix-script-runner/issues/35

**Pull request merged**